OpenRA.Mods.D2k/Traits/SpiceBloom.cs

```diff
@@ -74,10 +81,18 @@ public SpiceBloom(ActorInitializer init, SpiceBloomInfo info)
 			respawnTicks = self.World.SharedRandom.Next(info.RespawnDelay[0], info.RespawnDelay[1]);
 			growTicks = self.World.SharedRandom.Next(info.GrowthDelay[0], info.GrowthDelay[1]);
 			anim.Animation.Play(info.GrowthSequences[0]);
+
+			spawnCell = self.World.Map.CellContaining(self.CenterPosition);
 		}
 
 		public void Tick(Actor self)
 		{
+			if (!self.World.Map.Contains(spawnCell))
```

Comment:
It would be better to use `self.Location` instead of caching `self.World.Map.CellContaining(self.CenterPosition)`.  The latter will break if the actor moves for whatever reason.